File: app/app/src/main/java/com/resourceit/app/views/StatementsFragment.java
package com.resourceit.app.views;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.resourceit.app.Adaptera.StatmentsAdapter;
import com.resourceit.app.R;
import com.resourceit.app.holders.StatmentHolder;
import com.resourceit.app.models.StatmentModel;

import java.util.ArrayList;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class StatementsFragment extends Fragment {

    @BindView(R.id.statments)
    RecyclerView statments;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_statements, container, false);
        ButterKnife.bind(this, view);
        statments.setHasFixedSize(true);
        LinearLayoutManager mLayoutManager = new LinearLayoutManager(getActivity());
        statments.setLayoutManager(mLayoutManager);

        StatmentsAdapter mAdapter = new StatmentsAdapter(new ArrayList<StatmentHolder>(0));
        statments.setAdapter(mAdapter);
        for(int i = 1;i<=50;i++){
            StatmentModel st = new StatmentModel();
            st.setDate("1111-11-11");
            st.setTitle("khalid "+i);
            st.setDesc("teste desc "+i);
            st.setValue(i*0.3);
            mAdapter.insertItem(st);
        }
        return view;
    }
}

File: app/src/main/java/com/solinftec/desafiosantander_rafaelpimenta/view/fragment/StatementsFragment.kt
package com.solinftec.desafiosantander_rafaelpimenta.view.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.BindingAdapter
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.solinftec.desafiosantander_rafaelpimenta.R
import com.solinftec.desafiosantander_rafaelpimenta.databinding.StatementsFragmentBinding
import com.solinftec.desafiosantander_rafaelpimenta.model.UserAccount
import com.solinftec.desafiosantander_rafaelpimenta.util.BindingAdapters
import com.solinftec.desafiosantander_rafaelpimenta.util.Helper
import com.solinftec.desafiosantander_rafaelpimenta.view.adapter.StatementsAdapter
import com.solinftec.desafiosantander_rafaelpimenta.viewmodel.StatementsViewModel
import kotlinx.android.synthetic.main.statements_fragment.*
import java.text.NumberFormat
import java.util.*

class StatementsFragment : Fragment() {

    private lateinit var viewModel: StatementsViewModel
    private lateinit var binding: StatementsFragmentBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding =
            DataBindingUtil.inflate(inflater, R.layout.statements_fragment, container, false)

        return binding.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        viewModel = ViewModelProviders.of(this).get(StatementsViewModel::class.java)

        binding.viewModel = viewModel
        binding.rvStatements.adapter =
            StatementsAdapter(
                emptyList()
            )

        binding.rvStatements.layoutManager = LinearLayoutManager(activity)
        setUserInfo()


        binding.btnLogout.setOnClickListener {
            findNavController().popBackStack()
        }
    }


    override fun onStart() {
        super.onStart()

        viewModel.load()
    }

    private fun setUserInfo() {
       
        val user = arguments?.getSerializable("user") as UserAccount
        val balanceFormatted =
            NumberFormat.getCurrencyInstance(Locale("pt", "BR")).format(user.balance)
        val agencyFormatted = Helper().agencyMask(user.agency)
        
        viewModel.nameText.set(user.name)
        viewModel.accountText.set("${user.bankAccount} / $agencyFormatted")
        viewModel.balanceText.set(balanceFormatted)
  
    }
}